### DISCUSSION TITLE
How to deploy theia-ide in heroku or any other hosting platforms?

### DISCUSSION THREAD

**Discussion by PakkiPavan** [eclipse-theia/theia#9723] [Q&A]
I am trying to host theia ide in heroku using Dockerfile. But facing some issue while installing dependencies in Docker.
Can you help anyone how to host theia-ide in heroku or any other hosting platforms?
The [source code](https://github.com/PakkiPavan/theia-ide-2) that I am trying for theia-ide in heroku:

**Comment by vince-fugnitto** [comment]
@PakkiPavan thank you for the discussion, I don't believe there is any official guide or documentation on how to do so, but since your issues are related to dependencies in docker you may want to reference some of our docker images as part of the `theia-apps` repository and use it to start:
- https://github.com/theia-ide/theia-apps

These images are also available on `dockerhub`, for instance `theia-docker`:
- https://hub.docker.com/r/theiaide/theia/

In the past someone created a `theia-heroku` repository (however it is likely outdated) if you're interested in taking a look:
- https://github.com/jackvz/theia-ide-on-heroku

**Reply by PakkiPavan** [reply]
@vince-fugnitto Thanks for the reply. I have tried https://github.com/jackvz/theia-ide-on-heroku, it seems it is outdated as you told. I tried updating node version to 12.14.1, still facing below error:
```
 $ /home/theia/node_modules/.bin/theia start /home/project --hostname 0.0.0.0 --port 57970
 internal/modules/cjs/loader.js:1021
 return process.dlopen(module, path.toNamespacedPath(filename));
 ^
 Error: Error loading shared library libsecret-1.so.0: No such file or directory (needed by /home/theia/node_modules/keytar/build/Release/keytar.node)
 at Object.Module._extensions..node (internal/modules/cjs/loader.js:1021:18)
```
Can you help what is the issue in this?

**Reply by vince-fugnitto** [reply]
@PakkiPavan the repository is not maintained by the community as mentioned so there is likely to be issues. I think the best idea is to use the docker image examples we have as the dependencies like `libsecret` are handled. The `heroku` specific stuff is likely to remain the same:

- https://github.com/theia-ide/theia-apps/blob/master/theia-docker/Dockerfile

**Reply by PakkiPavan** [reply]
@vince-fugnitto I have tried https://github.com/theia-ide/theia-apps/blob/master/theia-docker/Dockerfile
Facing below error while running using Docker
Can you help what is the issue in this?
```
$ /home/theia/node_modules/.bin/theia build
Failed to resolve module: filenamify
Error: webpack exited with an unexpected signal: SIGKILL.
    at ChildProcess.<anonymous> (/home/theia/node_modules/@theia/application-manager/lib/application-process.js:60:28)
    at ChildProcess.emit (events.js:315:20)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
error Command failed with exit code 1.
```

**Reply by vince-fugnitto** [reply]
How did you reproduce the error, which command did you execute?

**Reply by PakkiPavan** [reply]
I have followed the below steps:
1. cloned https://github.com/theia-ide/theia-apps/blob/master/theia-docker/
2. In docker container, while running ```yarn theia build```, getting below error
```
Failed to resolve module: filenamify
Error: webpack exited with an unexpected signal: SIGKILL.
    at ChildProcess.<anonymous> (/home/theia/node_modules/@theia/application-manager/lib/application-process.js:60:2
```

**Reply by vince-fugnitto** [reply]
You can follow the documentation on how to build, and run the images:
- https://github.com/theia-ide/theia-apps/blob/master/CONTRIBUTING.md#theia-docker
